src/cli/commands/init.js

```diff
@@ -77,6 +77,7 @@ export async function run(config: Config, reporter: Reporter, flags: Object, arg
       key: 'repository',
       question: 'repository url',
       default: extractRepositoryUrl(repository),
+      isShorthand: /^[a-zA-Z0-9-_]*\/[a-zA-Z0-9-_]+$/,
```

Comment:
Let's reuse the existing matcher from https://github.com/yarnpkg/yarn/blob/master/src/resolvers/exotics/github-resolver.js#L17